v8_c_api/src/v8_c_api.cpp

```diff
@@ -306,6 +304,356 @@ v8_pd_list* v8_PDListCreate(v8::ArrayBuffer::Allocator *alloc) {
+    v8::Local<v8::String> message = (
+        stringView.is8Bit()
+          ? v8::String::NewFromOneByte(isolate, reinterpret_cast<const uint8_t*>(stringView.characters8()), v8::NewStringType::kNormal, length)
+          : v8::String::NewFromTwoByte(isolate, reinterpret_cast<const uint16_t*>(stringView.characters16()), v8::NewStringType::kNormal, length)
+      ).ToLocalChecked();
+    v8::String::Utf8Value result(isolate, message);
+    return *result;
+}
+
+class v8_inspector_channel_wrapper final: public v8_inspector::V8Inspector::Channel {
+public:
+    explicit v8_inspector_channel_wrapper(
+        v8::Isolate *isolate,
+        const InspectorOnResponseCallback &onResponse,
+        const InspectorUserDataDeleter &onResponseUserDataDeleter
+    );
+
+    void sendResponse(int callId, std::unique_ptr<v8_inspector::StringBuffer> message) override;
+    void sendNotification(std::unique_ptr<v8_inspector::StringBuffer> message) override;
+    void flushProtocolNotifications() override;
+
+    void setIsolate(v8::Isolate *isolate);
+
+    void setOnResponseCallback(
+        const InspectorOnResponseCallback &callback,
+        const InspectorUserDataDeleter &onResponseUserDataDeleter
+    );
+
+private:
+    v8::Isolate* isolate_;
+    InspectorOnResponseCallback onResponse_;
+    InspectorUserDataDeleter onResponseUserDataDeleter_;
+};
+
+
+v8_inspector_channel_wrapper::v8_inspector_channel_wrapper(
+    v8::Isolate *isolate,
+    const InspectorOnResponseCallback &onResponse,
+    const InspectorUserDataDeleter &onResponseUserDataDeleter
+)
+  :
+    isolate_(isolate),
+    onResponse_(onResponse),
+    onResponseUserDataDeleter_(onResponseUserDataDeleter)
+{}
+
+void v8_inspector_channel_wrapper::sendResponse(int callId, std::unique_ptr<v8_inspector::StringBuffer> message) {
+    const std::string response = convertToString(isolate_, message->string());
+    if (onResponse_) {
+        onResponse_(response);
+    }
+}
+
+void v8_inspector_channel_wrapper::sendNotification(std::unique_ptr<v8_inspector::StringBuffer> message) {
+    const std::string notification = convertToString(isolate_, message->string());
+    if (onResponse_) {
+        onResponse_(notification);
+    }
+}
+
+void v8_inspector_channel_wrapper::flushProtocolNotifications() {
+    // flush protocol notification
+}
+
+void v8_inspector_channel_wrapper::setIsolate(v8::Isolate *isolate) {
+    isolate_ = isolate;
+}
+
+void v8_inspector_channel_wrapper::setOnResponseCallback(
+    const InspectorOnResponseCallback &callback,
+    const InspectorUserDataDeleter &deleter
+) {
+    if (onResponseUserDataDeleter_) {
+        onResponseUserDataDeleter_();
+    }
+
+    onResponse_ = callback;
+    onResponseUserDataDeleter_ = deleter;
+}
+
+class v8_inspector_client_wrapper final: public v8_inspector::V8InspectorClient {
+public:
+    explicit v8_inspector_client_wrapper(
+        v8::Platform *platform,
+        const v8::Local<v8::Context> &context,
```

Comment:
I do not like saving a Local value here as it will only leave as long as the local scope. I also see that you only use it to get the isolate, so why not just passing the isolate from the first place?

Reply:
Ohh ok, I mistakenly thought we are saving the local handler. :+1: